File: random_problems/teste/5.cpp
#include <iostream>
#include <string>
#include <bits/stdc++.h>
using namespace std;

int main() {
    // 2 1 1 1
    std::string tf1, vqr1, tf2, vqr2; cin >> tf1 >> vqr1 >> tf2 >> vqr2;
    long long taxa1 = stod(tf1) * 1e5;
    long long valor1 = stod(vqr1) * 1e5; // valor por km rodado

    long long taxa2 = stod(tf2) * 1e5;
    long long valor2 = stod(vqr2) * 1e5;


    if (taxa1 == taxa2 && valor1 == valor2) {
        cout << "Tanto Faz" << endl;
        return 0;
    }

    if (taxa1 == taxa2) {
        cout << (valor1 > valor2 ? "Empresa 2" : "Empresa 1") << endl;
        return 0;
    }

    if (valor1 == valor2) {
        cout << ( taxa1 < taxa2 ? "Empresa 1" : "Empresa 2") << endl;
        return 0;
    }

    if (taxa1 > taxa2 && valor1 > valor2) {
        cout <<  "Empresa 2" << endl;
        return 0;
    }

    if (taxa2 > taxa1 && valor2 > valor1) {
        cout <<  "Empresa 1" << endl;
        return 0;
    }

    /* cout << "taxa1 -> " << taxa1 << endl; */
    /* cout << "taxa2 -> " << taxa2 << endl; */

    /* cout << "valor1 -> " << valor1 << endl; */
    /* cout << "valor2 -> " << valor2 << endl; */
    double igual_em = (taxa2 - taxa1) / (valor1 - valor2);

    cout << "igual_em _> " << igual_em << endl;

    std::string menor = (taxa2 - taxa1 > 0 ? "1" : "2");
    std::string maior = (menor == "1" ? "2" : "1");

    std::string num_text = std::to_string(igual_em);
    std::string resp = num_text.substr(0, num_text.find(".")+2);


    cout << ( "Empresa " + menor + " quando a distância é " + resp + ", Tanto faz quando a distância é = " + resp + ", Empresa " + " quando a distância é > " + resp ) << endl;
}

File: random_problems/teste/5.py
def escolhe_taxi(tf1,vqr1,tf2,vqr2):
    taxa1 = float(tf1)
    taxa2 = float(tf2)
    valor1 = float(vqr1)
    valor2 = float(vqr2)
    
    if taxa1 == taxa2 and valor1 == valor2:
        return "Tanto Faz"
    
    if taxa1 == taxa2:
        if valor1 > valor2:
            return "Empresa 2"
        else:
            return "Empresa 1"
    
    if valor1 == valor2:
        if taxa1 < taxa2:
            return "Empresa 1"
        else:
            return "Empresa 2"
    
    if taxa1 > taxa2 and valor1 > valor2:
        return "Empresa 2"
    
    if taxa2 > taxa1 and valor2 > valor1:
        return "Empresa 1"
        
    igual_em = (taxa2 - taxa1) / (valor1 - valor2)
    
    menor = ""
    maior = ""
    
    if taxa2 - taxa1 > 0:
        menor = 1
        maior = 2
    else:
        menor = 2
        maior = 1

    if igual_em.is_integer():
        return f"Empresa {menor} quando a distância < {igual_em:.1f}, Tanto faz quando a distância = {igual_em}, Empresa {maior} quando a distância > {igual_em}"

print(escolhe_taxi(2.5, 1.0, 5.0, 0.75))
print(escolhe_taxi(2, 1, 1, 3))
